libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -62,27 +63,21 @@ void DataFlowAnalyzer::operator()(ExpressionStatement& _statement)
 	if (auto vars = isSimpleStore(StoreLoadLocation::Storage, _statement))
 	{
 		ASTModifier::operator()(_statement);
-		set<YulString> keysToErase;
-		for (auto const& item: m_storage.values)
-			if (!(
-				m_knowledgeBase.knownToBeDifferent(vars->first, item.first) ||
-				m_knowledgeBase.knownToBeEqual(vars->second, item.second)
-			))
-				keysToErase.insert(item.first);
-		for (YulString const& key: keysToErase)
-			m_storage.eraseKey(key);
-		m_storage.set(vars->first, vars->second);
+		cxx20::erase_if(m_storage, [&](auto const& entry) {
+			return !(
+				m_knowledgeBase.knownToBeDifferent(vars->first, entry.first) ||
+				m_knowledgeBase.knownToBeEqual(vars->second, entry.second)
```

Comment:
Yeah, I thought about it, but since it was like this before, I just kept it - I'm not entirely sure which way it's easier to make sure the condition makes sense myself :-). But yeah, I can deMorgen-flip it.